images/node/Dockerfile-debian

```diff
@@ -0,0 +1,51 @@
+###############################################################
+# DOCKER IMAGE - Debian based Nodejs Application (Express.js) #
+###############################################################
+
+FROM debian:latest
+
+# Install any dependencies for building & commands to use
+RUN apt-get update && apt-get install -y python3 make curl
+
+# Set verions of using packages
+ARG NODE_VERSION=18
+
+# Install NodeJS
+RUN apt-get update -yq \
+    && apt-get install -yq ca-certificates curl gnupg \
+    && mkdir -p /etc/apt/keyrings \
+    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
+    && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_VERSION.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
+    && apt-get update -yq \
+    && apt-get install nodejs -yq \
+    && npm install -g npm
+
+# Set the working directory within the container
+WORKDIR /app
+
+# Copy both package.json and package-lock.json
+COPY package*.json ./
+
+# Copy the application code into the container
+COPY . /app
+
+# Install application dependencies
+RUN npm install
+
+# Create the "limitedaccessgroup" group
+RUN groupadd limitedaccessgroup
+
+# Create a non-root user named "limitedaccessaccount" with a specific user ID (e.g., 1050)
+RUN adduser --disabled-password --gecos '' --uid 1050 limitedaccessaccount
+
+# Add the "limitedaccessaccount" to the "limitedaccessgroup" group
+RUN usermod -a -G limitedaccessgroup limitedaccessaccount
+
+# Set non root user
+USER 1050
+
+# Expose the port that the Express.js application will run on (default is 3000)
+EXPOSE 3000
+
+# Define the command to start the Express.js application
+CMD node app.js
```

Comment:
lets use entrypoint instead of cmd, though interested to get @brandan-schmitz 's thoughts.